linkerd/transport-header/proto/header.proto

```diff
@@ -8,4 +8,18 @@ message Header {
 
   // An optional hostname. Intended for gateway forwarding.
   string name = 2;
+
+  // The session protocol, if one is known. When no protocol is specified, the
+  // connection is handled opaquely.
+  SessionProtocol session_protocol = 3;
+}
+
+message SessionProtocol {
+  message Http1 {}
+  message H2 {}
```

Comment:
nit/TIOLI: it bothers me slightly that we have "http 1" and "h2", can we make these consistent? not a big deal...

Reply:
...i guess we already use this naming scheme with `http::Version`, so it's whatever i guess.